src/condition.cpp

```diff
@@ -307,6 +307,28 @@ void conditional_t<T>::set_has_perception( const JsonObject &jo, const std::stri
     };
 }
 
+template<class T>
+void conditional_t<T>::set_has_hp( const JsonObject &jo, const std::string &member, bool is_npc )
+{
+    int_or_var iov = get_int_or_var( jo, member );
+    cata::optional<bodypart_id> bp;
+    optional( jo, false, "bodypart", bp );
+    condition = [iov, bp, is_npc]( const T & d ) {
+        bodypart_id bid = bp.value_or( bodypart_str_id::NULL_ID() );
+        if( bid == bodypart_str_id::NULL_ID() ) {
+            // If bp is null, check if reason string represents a valid bodypart_id
```

Comment:
Do you hope to refactor this repeated block? Seems like at least part of it can be pulled out into a helper of some kind.

Reply:
That's the plan. There's an established pattern in the condition functions that I'd be breaking either way. But this is a temporary solution until I work out something cleaner.

By the way, I've got some exciting developments! Check out https://github.com/CleverRaven/Cataclysm-DDA/pull/54693/commits/a758fd3ad9812dcf9b4171acd7d0cf8945152214, I was able to completely JSONify `bp_status_text` (as in, the widget text and colours are no longer generated with a `display` function, instead using only conditional phrases). The next step is to migrate a bunch of other `display` candidates (ex: radiation badge) to JSON-only definitions.

Reply:
Oh, fantastic! That is great to hear. Can’t believe how fast things have been moving in the world of widgets lately. The future is bright.